### PR TITLE
refactor: Fix clang-tidy issues in headers

### PR DESCRIPTION
**Pull request by coder3101**
## Description

While building the header files are not checked by clang-tidy, but my IDE linted these issues in header files and they have been all fixed in this refactor PR.

## Related Issues / Pull Requests
N/A

## Motivation and Context
So that header files are also free from clang-tidy issues.

## How Has This Been Tested?
Arch on KDE

## Checklist

- [x] If the key of a setting is changed, the `old` attribute is updated or it is resolved in SettingsUpdater.
- [x] If there are changes of the text displayed in the UI, they are wrapped in `tr()` or `QCoreApplication::translate()`.
- [x] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [x] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

## Additional text
N/A

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
+Checks:                 '-*,bugprone-*,cppcoreguidelines-*,modernize-*,performance-*,readability-*,-modernize-use-trailing-return-type,-bugprone-suspicious-include,-cppcoreguidelines-pro-bounds-array-to-pointer-decay,-cppcoreguidelines-avoid-magic-numbers,-readability-magic-numbers,-cppcoreguidelines-owning-memory,-readability-braces-around-statements,-cppcoreguidelines-macro-usage,-cppcoreguidelines-avoid-c-arrays,-modernize-avoid-c-arrays,-readability-implicit-bool-conversion,-readability-implicit-bool-conversion,-cppcoreguidelines-pro-type-static-cast-downcast,-cppcoreguidelines-special-member-functions,-cppcoreguidelines-avoid-non-const-global-variables,-performance-no-automatic-move,-readability-redundant-access-specifiers'
```
Comment (coder3101) [inline .clang-tidy:2]: Do not check for redundant access specifiers. See [this](https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-redundant-access-specifiers.html).

File: src/Extensions/ClangFormatter.hpp [hunk 2]
```diff
@@ -37,17 +37,17 @@ class ClangFormatter : public CodeFormatter
                             MessageLogger *log, QObject *parent = nullptr);
 
   protected:
-    virtual QString settingKey() const override;
```
Comment (coder3101) [inline src/Extensions/ClangFormatter.hpp:40]: `virtual` is redundant if the function is marked `override`.

File: src/Extensions/CodeFormatter.hpp [hunk 3]
```diff
@@ -91,11 +91,13 @@ class CodeFormatter : public QObject
     QVariant getSetting(const QString &key) const;
 
   protected:
-    QCodeEditor *editor;
-    QString lang;
-    bool selectionOnly;
-    bool logOnNoChange;
-    int cursorPos, cursorLine, cursorCol, anchorPos, anchorLine, anchorCol;
+    QCodeEditor *editor;                             // NOLINT: cppguidelines do not support use of protected members
```
Comment (coder3101) [inline src/Extensions/CodeFormatter.hpp:94]: Protected data members is bad. See [this](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-protected)

**Review by ouuan — APPROVED** [review]
LGTM

**Pull request merged**